org.eclipse.scanning.points/src/org/eclipse/scanning/points/AbstractScanPointIterator.java

```diff
@@ -28,6 +52,66 @@ public void setPyIterator(Iterator<IPosition> pyIterator) {
 		this.pyIterator = pyIterator;
 	}
 
+	protected Iterator<IPosition> createSpgCompoundGenerator(Iterator<?>[] iterators, Object[] regions,
+			String[] regionAxes, PyObject[] mutators) {
+		JythonObjectFactory<PyObject> excluderFactory = ScanPointGeneratorFactory.JExcluderFactory();
+		@SuppressWarnings("rawtypes")
+		JythonObjectFactory<SerializableIterator> cpgFactory = ScanPointGeneratorFactory.JCompoundGeneratorFactory();
+		List<PyObject> pyRegions = Arrays.asList(regions).stream().map(r -> makePyRoi(r)).collect(Collectors.toList());
+		pyRegions = pyRegions.stream()
+				.filter(r -> r != null)
+				.map(r -> excluderFactory.createObject(r, new PyList(Arrays.asList(regionAxes))))
+				.collect(Collectors.toList());
+		@SuppressWarnings("unchecked")
+		Iterator<IPosition> cpgIterator = cpgFactory.createObject(iterators, pyRegions.toArray(), mutators);
+		return cpgIterator;
+	}
+
+	static {
+		roiDispatchMap.put(CircularROI.class, r -> ScanPointGeneratorFactory.JCircularROIFactory().createObject(
+				((CircularROI) r).getCentre(), ((CircularROI) r).getRadius()));
+		roiDispatchMap.put(EllipticalROI.class, r -> ScanPointGeneratorFactory.JEllipticalROIFactory().createObject(
+				((EllipticalROI) r).getPoint(), ((EllipticalROI) r).getSemiAxes(), ((EllipticalROI) r).getAngle()));
+		roiDispatchMap.put(LinearROI.class, r -> null); // not supported
+		roiDispatchMap.put(PointROI.class, r -> ScanPointGeneratorFactory.JPointROIFactory().createObject(
+				((PointROI) r).getPoint()));
+		roiDispatchMap.put(PolygonalROI.class, r -> {
+			PolygonalROI p = (PolygonalROI) r;
+			double[] xPoints = new double[p.getNumberOfPoints()];
+			double[] yPoints = new double[p.getNumberOfPoints()];
+			for (int i = 0; i < xPoints.length; i++) {
+				PointROI point = p.getPoint(i);
+				xPoints[i] = point.getPointX();
+				yPoints[i] = point.getPointY();
+			}
+			return ScanPointGeneratorFactory.JPolygonalROIFactory().createObject(xPoints, yPoints);
+		});
+		roiDispatchMap.put(RectangularROI.class, r -> ScanPointGeneratorFactory.JRectangularROIFactory().createObject(
+				((RectangularROI) r).getPoint(), ((RectangularROI) r).getLength(0), ((RectangularROI) r).getLength(1),
+				((RectangularROI) r).getAngle()));
+		roiDispatchMap.put(SectorROI.class, r -> ScanPointGeneratorFactory.JSectorROIFactory().createObject(
+				((SectorROI) r).getPoint(), ((SectorROI) r).getRadii(), ((SectorROI) r).getAngles()));
+	}
+
+	protected static PyObject makePyRoi(Object region) {
+		IROI roi = null;
+		if (region instanceof ScanRegion<?>) {
+			region = ((ScanRegion<?>) region).getRoi();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "region". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1226)